src/libs/EmojiUtils.ts

```diff
@@ -569,6 +569,120 @@ function getSpacersIndexes(allEmojis: EmojiPickerList): number[] {
     return spacersIndexes;
 }
 
+/**
+ * Split string into plain text and emojis array with attention to:
+ * Surrogate pairs (combined emojis including flags)
+ * Modifiers (different skin tones and emoji variations)
+ * @param text
+ */
+
+// Surrogate pairs (combined emojis)
+const highSurrogateStart = 0xd800;
+const highSurrogateEnd = 0xdbff;
+const lowSurrogateStart = 0xdc00;
+const zeroWidthJoiner = 0x200d;
+
+// Regional indicator symbols (flags)
+const regionalIndicatorStart = 0x1f1e6; // 1st letter of a two-letter country code
+const regionalIndicatorEnd = 0x1f1ff; // Last letter of a two-letter country code
+
+// Fitzpatrick scale modifiers (skin tone)
+const fitzpatrickScaleStart = 0x1f3fb; // Type 1 (represents light skin tone)
+const fitzpatrickScaleEnd = 0x1f3ff; // Type 6 (represents dark skin tone)
+
+// Variation selectors (specific variations in the presentation of other characters)
+const variationModifierStart = 0xfe00; // Request text presentation of emoji
+const variationModifierEnd = 0xfe0f; // Indicate that the character should be displayed as an emoji
+
+const codePointFromSurrogatePair = (pair: string) => {
```

Comment:
let's use the `function` keyword instead of arrow functions, just because it's the more common pattern in this codebase